_posts/2020-11-04-ggplot2.md

```diff
@@ -0,0 +1,92 @@
+
+## Adding the Geoms
+# 1. Bar Plots
+
+There are many types of *geoms* in `ggplot2`, but the syntax for all of them is
+the same. Let us start with a common and very useful type of visualization: the
+bar graph. There are two *geoms* for this: *geom_bar* and *geom_col*. We will
+go over both of these in the following code chunk since they use the same parameters. All the default values will be included
+for each parameter and elaborated on.
+
+```{r make-bar-plot}
+# Taking the same data set from the last code chunk
+ggplot(nice_data, aes(x = "dependent value", y = "independent value")) +
+geom_col(mapping = "NULL",
+data = "NULL", stat = "count", position = stack,
+width = NULL, na.rm = FALSE, orientation = NA, show.legend = NA, inherit.aes =
+TRUE)
+```
+The **data** inside the *geom* refers to your dataset from before
+and if set to *NULL*, will automatically refer to your dataset
+from before. The **stat** records the count of each *x* or *y* value depending on your preferred orientation and is best suited for histograms (note that there is a separate *geom_histogram()* for this as well). The **width** refers to the width of each bar in the plot, you might have to adjust this depending on crowded your axis is. A good way to determine this is simply if you can match each data value with its column. If you cannot, then go for a smaller width so you can match these values. **na.rm** refers to missing values: if set to *TRUE*, they are just removed, if set to *FALSE*, they are removed but with a warning. **orientation** refers to the axes the dependent and independent values are on: if set to *NA* `ggplot2` will figure out which is which on its own. This parameter can be set to x or y if this fails for any reason. **show.legend** refers to the legend for your dataset: when set to *NA*, it automatically includes anything that is visualized in the graph: most likely *x* and *y* will be described here. **inherit.aes** refers to using the `aes` values from the first section of the code. **position** will either be "stack" where x values are on top of each other, or "dodge" where they are kept separate.
+
+# 2. Box and Whisker Plots
+
+Box and whisker plots are especially useful for scientific studies: it allows you not to just see data, but also the margin of error and how wide it is. A lot of parameters here
+are the same for *geom_col* and *geom_bar*, but *geom_boxplot* will need further elaboration about the lower and upper quartiles of data. Again, let's do this in a code
+chunk:
+
+```{r box-and-whisker-plot}
+#include default parameters here
```

Comment:
Consider doing a review of how you show comments across all of your code chunks in order to maintain style consistency (e.g. capitalize the first letter, include a space after the hash, etc.)